### DISCUSSION TITLE
Refactor CI to run vcstool outside the Docker image

### DISCUSSION THREAD

**Discussion by EzraBrooks** [space-ros/docker#242] [Ideas]
using `vcs`, or other methods of using Git, inside your Dockerfile causes all sorts of disk space and caching shenanigans.

**Comment by Bckempa** [comment]
Preliminarily nominating this for next release as part of a planned revision of the existing docker files. If it doesn't make it as part of that clean-up effort it can be descoped.

**Comment by xfiderek** [comment]
Which CI workflow and docker image are we refering to? Is it this one:
https://github.com/space-ros/docker/blob/main/.github/workflows/earthly-build.yaml

**Comment by EzraBrooks** [comment]
No, this is about removing invocations of `vcs` from inside `Dockerfile`s throughout the repository and instead calling it externally to avoid filling the Docker image up with Git metadata and having idempotency issues
